src/svd.cc

```diff
@@ -316,17 +328,23 @@ void svd(
             redistribute(U1d_row_cyclic, U1d, opts);
 
             // First, U = U2 * U ===> U1d = U2 * U1d
+            Timer t_unmtr_hb2st;
             unmtr_hb2st( Side::Left, Op::NoTrans, U2, U1d, opts );
+            timers[ "svd::unmtr_hb2st" ] = t_unmtr_hb2st.stop();
 
             // Redistribute U1d into U
             redistribute(U1d, Uhat, opts);
 
             // Second, U = U1 * U ===> U = Ahat * U
+            Timer t_unmbr_ge2tb;
             unmbr_ge2tb( Side::Left, Op::NoTrans, Ahat, TU, Uhat, opts );
+            timers [ "svd::unmbr_ge2tb" ] = t_unmbr_ge2tb.stop();
             if (qr_path) {
                 // When initial QR was used.
                 // U = Q*U;
+                Timer t_unmqr;
                 unmqr( Side::Left, slate::Op::NoTrans, A, TQ, U, opts );
+                timers [ "svd::unmqr" ] = t_unmqr.stop();
             }
```

Comment:
How would moving the timer outside the `if (qr_path) { ... }`  or `if (lq_path) { ... }` simplify the tester?

Reply:
If `timer[ "svd::geqrf" ]` is not always defined, the tester needs to check before accessing it. If the tester can rely on it always being defined, it doesn't need to check first.

Reply:
I see, that makes sense. Are you wanting to move every timer outside of `if (qr_path) { ... }` and `if (lq_path) { ... }` for `svd.cc`? There are 4 total conditional statements for `qr_path` and `lq_path`.  Just want to be clear on this before making the change. The timers that would be moved out would be:

- `geqrf`
- `gelqf`
- `unmqr`
- `unmlq`

Reply:
For `geqrf` and `gelqf`, it seems hard to pull the timers out of the `if` blocks, since there's an `else`. Maybe just above that `if` block (line 146) set
```
timers[ "svd::geqrf" ] = 0;
timers[ "svd::gelqf" ] = 0;
```
For `unmqr` and `unmlq`, you can put the timer around the whole `if` block, it seems.

Reply:
Got it, making the changes now to `svd.cc`. Another option to keep inline with your suggestion of setting:
```
timers[ "svd::geqrf" ] = 0;
timers[ "svd::gelqf" ] = 0;
```
Is that we could just set all timers to be `=0` at the start. That possibly may be over doing it, but it keeps all initialization together.

Reply:
I was surprised to see in the tester that `timers[ "svd::tb2db" ]` (with the typo) didn't throw an error. It seems `std::map` will insert a 0 value when accessing a non-existent entry. Using `timers.at( "svd::tb2db" )` would throw an error. The convention in `std` containers is `foo[ index ]` doesn't check bounds, while `foo.at( index )` checks bounds and throws errors. So we don't really need to initialize everything to 0, but perhaps in this case it's not a bad idea.

Reply:
That is a fairly odd choice to insert a value of 0 when accessing a non-existent entry. Should throw an error if it is not within the `std::map` in my opinion or if wanting to keep with the idea of inserting a value it should be NA or NAN.

Reply:
Technically, since std containers are templated on type `T`, it inserts an element `T()`. In the case of doubles, `double()` is 0. For a map of int, `int()` is also 0. For a map of strings, `string()` is an empty string. If we had a map of `slate::Matrix`, `Matrix()` is an empty matrix. It's a little odd that it doesn't throw errors for non-existent entries, but fits with the distinction between `[ ]` not throwing, and `at( )` throwing errors for classes like `std::vector`.

Perl does the same thing, that `foo{ "bar" }` is 0 or an empty string, while in Python, `foo[ "a" ]` throws a KeyError.